modules/vmseries/main.tf

```diff
@@ -86,7 +86,14 @@ resource "azurerm_virtual_machine" "this" {
   }
 
   os_profile_linux_config {
-    disable_password_authentication = false
+    disable_password_authentication = var.password == null ? true : false
```

Comment:
I think that `password` var should no longer be a required one (ie. have a default set to `null`). Additionally, it would be good to document the fact that if password auth is disabled, then at least one key has to be specified (https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/virtual_machine#ssh_keys).

Reply:
good point, I've set `null` as the default for the `password` variable.

Reply:
implemented